src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -704,6 +721,10 @@ private static bool TryGetIntQueryParameterValue(NameValueCollection queryString
                 {
                     return await client.WaitForCompletionOrCreateCheckStatusResponseAsync(request, id, timeout.Value, pollingInterval.Value);
                 }
+                else if (timeout.HasValue && client is DurableClient durableClient && durableClient.DurabilityProvider.SupportsPollFreeWait)
+                {
+                    return await client.WaitForCompletionOrCreateCheckStatusResponseAsync(request, id, timeout.Value, timeout.Value);

```

Comment:
Can we document how this else-if is different from the pre-existing if-statement above? They seem quite similar. 

In fact, there might be a refactor we we only check for `if (timeout.HasValue)` and inside of that we switch the final parameter to `WaitForCompletionOrCreateCheckStatusResponseAsync` depending on some differentiating condition.

I do not feel strongly about a refactor like that, but I would recommend a quick comment explaining the difference. Thanks!

Reply:
I changed the logic for this a bit, and added a comment.